Assets/Scripts/Data/PickupPup/Currency/DogFoodData.cs

```diff
@@ -10,9 +10,16 @@
 [System.Serializable]
 public class DogFoodData : CurrencyData
 {
-    public DogFoodData(int initialAmount) : base(initialAmount)
+    public DogFoodData(int initialAmount, bool isSpecial) : base(initialAmount)
```

Comment:
I think `DogFoodData` should simply take the `CurrencyType` as an argument (makes it easier to extend). Possible also worth creating a separate enum `DogFoodType` so that all dog food's are the same basic `CurrencyType`.